### ISSUE TITLE
File gets uploaded locally but when trying on deployed version it gets stuck on redirecting 

### ISSUE THREAD

**Issue by rishav887** [joschan21/quill#33]
File gets uploaded locally but when trying on deployed version it gets stuck on redirecting 

on the network tab i see - i have deployed it on railway.app

[
    {
        "error": {
            "message": "NOT_FOUND",
            "code": -32004,
            "data": {
                "code": "NOT_FOUND",
                "httpStatus": 404,
                "path": "getFile"
            }
        }
    }
]

**Comment by willbrandin** [comment]
I have this issue too. I'm not seeing the new file added in the database either. 
I wonder if there's an issue in the `onUploadComplete` function in `src/app/api/uploadthing/core.ts`

**Comment by engmsaleh** [comment]
I have the same issue, Any updates or solutions?

**Comment by rishav887** [comment]
Versel works fine for me

**Comment by engmsaleh** [comment]
@willbrandin Yes, I have encountered the same issue, for some reason, it seems that `onUploadComplete` is not being fired in a production environment

**Comment by husmaret** [comment]
Could be an authentication issue, as I also get a `401` error just before the `404`. Meaning the `middeware` before `onUploadComplete` might return an Unauthorized. But I also have not found a solution yet.

Edit: Ok, so apparently not an Authentication issue. In 'Dev' mode, uploadthing simulates a webhook, to notify the server, when the fileupload is finished. This is not done in 'production' environment. Still do not know how to solve it, but there is an issue on the uploadthing github: https://github.com/pingdotgg/uploadthing/issues/474

**Comment by m3phist** [comment]
I finally got the file upload running on production by installing the new version of uploadthing@6.1.0 and @uploadthing/react@6.0.2. Make sure to push and test directly on Vercel, not in the local npm build. Anyway, Uploadthing warns not to test callbacks on localhost. That is how I finally got it to work.

_[UT] [WARN] You are using a localhost callback url in production which is not supported. Read more and learn how to fix it here: https://uploadthing.com/faq#my-callback-runs-in-development-but-not-in-production_

**Comment by alvinpeter9** [comment]
The uploadthing documentation has not been updated to reflect the new update. @m3phist can you show us where or what you changed in your code?

**Comment by gptsparky** [comment]
I had the same issue, file uploading didn't work when deployed. It was stuck at the upload box loading. Looking at the logs, the api/trpc/getFile endpoint was being polled every few seconds and never stopped (also error 404).

Afterward, I noticed the file was successfully uploaded to UploadThing but it was NOT added to the database (thus the infinite polling), so the issue was the code to add the file to the database wasn't running (the code in the callback _onUploadComplete_ in _src/app/api/uploadthing/core.ts_).

Turns out Vercel was the issue, it allowed a maximum of a few seconds of computing at a time so the uploaded file was never added to the database as the code to do it timed out. Thus, the infinite polling.

TLDR; the solution is to deploy elsewhere like Netlify or upgrade from the Hobby plan on Vercel.

**Comment by seancaleb** [comment]
@gptsparky I'm having the same issue as you when I deployed my express app on railway. I am hosting my client in Vercel and currently using `Preview Deployments`. I've already did what UploadThing docs say about turning off `Vercel Authentication` [Vercel preview deployments](https://docs.uploadthing.com/faq#my-callback-runs-in-development-and-production-but-not-in-vercel-preview-deployments), unfortunately, it still didn't work for me.

It uploads the file to UploadThing but the `onUploadComplete` callback doesn't trigger which causes the loading on my client to not stop.

Upon viewing the logs from railway I saw this kind of error.

![image](https://github.com/joschan21/quill/assets/74448818/c461eb80-2400-48e3-a765-163263e9ad62)

**Comment by Selvagokul** [comment]
![Screenshot (413)](https://github.com/joschan21/quill/assets/121251457/c4986ace-875c-4ab6-8648-7b50f9b39fbe)
in live file is not uploaded give me any solution
